### ISSUE TITLE
GLM-5 / 5.1 support and optimization plan

### ISSUE THREAD

**Issue by xwu-intel** [vllm-project/vllm-xpu-kernels#224]
The issue serves as tracking GLM-5/5.1 XPU support and optimizations for vLLM.

## Overview
GLM-5 officially released BF16 weights (zai-org/GLM-5, 1.51TB) and FP8 weights (zai-org/GLM-5-FP8, 756GB). GLM-5.1 shares the same architecture with GLM-5 and is an advancement of GLM-5 for agentic engineering. It also released BF16 weights (zai-org/GLM-5.1, 1.51TB) and FP8 weights (zai-org/GLM-5.1, 756GB). 
Due to the similar architecture of DeepSeek V3.2 and GLM-5/5.1, we will reuse XPU MLA Sparse backend enabled for DeepSeek v3.2 #154. Besides Sparse MLA Attention, there are also common layers needed. The following features we supported will not only be used by DeepSeek v3.2 and GLM-5 but also benefit other models in general. 

## FP8 Quantization
For the original BF16 1.51 TB weights, it's not possible to load the full weights into 8 XPU cards. Quantization is necessary. Considering practical usage, we will focus on supporting (zai-org/GLM-5-FP8, 756GB) and (zai-org/GLM-5.1-FP8, 756GB) with TP for multi-card servers. 

### Dynamic Quantization
By default, when loading zai-org/GLM-5-FP8 weights with quantization="fp8", it will go through W8A8 path with block fp8 quantization. Currently, the Triton fallback paths are used. We will enable the following layers for native W8A8 FP8 quantization.

| Layer | Class | Kernels | Plan | PR |
| :--- | :--- | :--- | :--- | :--- |
| **FP8 Linear** | `W8A8BlockFp8LinearOp` | `w8a8_triton_block_scaled_mm` (Triton)<br>`cutlass_scaled_mm` | Implement block `scaled_mm` with oneDNN in `vllm-xpu-kernels` in the short term; migrate to PyTorch `scaled_mm` in the long term. | kernel: #173 <br> integration: https://github.com/vllm-project/vllm/pull/39968 |
| **FP8 MoE** | `TritonExperts`<br>`XPUExpertsFp8` | `fused_moe_kernel` (Triton)<br>`xpu_fused_moe` | Implement `xpu_fused_moe` for W8A8.  | TBA |

DeepSeek V3.2 and GLM-5 default block fp8 weight and activation format:

| Weight quantization | Activation quantization | Meaning |
| :--- | :--- | :--- | 
| kFp8Static128BlockSym | kFp8Dynamic128Sym | [DeepSeek V3.2, GLM5] static 128x128 weight scaling with dynamic 1x128 activation scaling. |

### Static Quantization
While dynamic quantization is the default method of quantization, static quantization is the alternative way to lower runtime overhead and deliver better performance. We will investigate the use of [vllm-project/llm-compressor](https://github.com/vllm-project/llm-compressor) with INC integration for static quantization.

## MXFP8 Quantization
MXFP8 uses block-level scaling (one scale factor for every 32 elements), it captures the dynamic range of BF16 much more accurately than standard FP8. MXFP8 is natively supported by the next gen XPU hardware. 
- [x] MXFP8 Linear: https://github.com/vllm-project/vllm/pull/38707
- [ ] MXFP8 MoE: TBA

## Performance kernels integration
- [ ] https://github.com/vllm-project/vllm/pull/37888
- [ ] fp8_mqa_logits and fp8_paged_mqa_logits xpu integration https://github.com/vllm-project/vllm-xpu-kernels/pull/299
- [ ] Leverage vLLM IR for integration (direction) e.g. https://github.com/vllm-project/vllm/issues/38745

## Enable torch.compile for XPU MLA Sparse backend

We need to resolve the following issues for enabling torch.compile for XPU MLA Spare backend

- [X] Fix warmup issue https://github.com/vllm-project/vllm/pull/37523
- [x] Fix fusion passes for XPU (rms_norm_static_fp8_quant) https://github.com/vllm-project/vllm/pull/39789

## Parallelism

We will verify Tensor Parallelism as initial step and expand to TP+DP+EP eventually.
